docs/package_firewall/faq.md

```diff
@@ -0,0 +1,56 @@
+# Frequently Asked Questions
+
+## What do I do when a dependency fails analysis?
+
+If a package's installation gets blocked by the firewall, possible malicious
+code execution will have been blocked, so there's no immediate need for action.
+
+Should you determine the policy violation to be a false-positive, you can add an
+exception by navigating to [the firewall overview], selecting the group
+associated with the project, then clicking on the "Allow Package" button in the
+top-right.
+
+[the firewall overview]: https://app.phylum.io/firewall
+
+## Why are some packages violating a policy not being blocked?
+
+The package firewall acts as a proxy between the local package manager and the
+official registry. As a result only packages being retrieved from the official
+registry will be analyzed.
+
+Common scenarios where a package will not be analyzed are the usage of a cache
+between the package firewall and the local client (like Artifactory, or Nexus),
+and the installation from the local registry cache. While the initial
+installation would be analyzed by the package firewall, once cached it will not
+be re-analyzed.
+
+Currently some ecosystems also bypass analysis when a package is installed
+directly from a lockfile. This is why new dependencies should always be
+installed with the package firewall in place, making sure that the lockfile
+contains no packages which violate your policy.
+
+## How do I deal with failures due to unprocessed packages?
+
+By default, the package firewall considers unprocessed packages as failures, to
+avoid running any code that hasn't been analyzed. Any unprocessed package is
+automatically staged for processing, so the safest option is always to wait
+until processing is complete.
+
+If you wish to allow certain or all packages to be considered successful for the
+purpose of the package firewall, please reach out to [Veracode][aviary-support].
```

Comment:
Why are you suggesting they reach out to Veracode and not suggesting they self-service using the `Allow Incomplete` button in the UI?

Reply:
I was unaware we had that button in the prod UI at this point. I'll change it.

Reply:
Would it also help here to explain how to set up a policy that automatically restricts/filters packages that are newer than a configurable timespan...say two weeks? Doing so should allow for a few benefits:

* The package will already be processed by Phylum
  * Greatly reduces the chance of waiting for complete analysis
  * Prevents opening the "gap" in coverage with the `Allow Incomplete` button
* Gives new packages/versions a "cool-off" period
  * It seems a lot of malware is discovered/triaged early on
  * The package/version has time to be removed from the related registry

Reply:
Such a policy would very likely get in the way of development and is much more likely to deny access to important security updates than safe people from malicious code. Additionally they would fail analysis, which would be opposite to the goal here, since it would just cause more packages to fail rather than less.